### PR TITLE
Updating network schemes; Adding related tests

### PR DESCRIPTION
**Pull request by EvgenyAgafonchikov**

### CODE REVIEW

**Comment by TianoMS** [comment]
@EvgenyAgafonchikov please update 2015-01-01/deploymentTemplate.json and 2014-04-01-preview/deploymentTemplate.json with your changes.

**Comment by TianoMS** [comment]
Can we have folks from NRP to review this? Also, we have another [PR](https://github.com/Azure/azure-resource-manager-schemas/pull/286) which also adds network schemas. Any connection between these two?

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 1]
```diff
@@ -180,6 +180,54 @@
...
                   { "$ref": "http://schema.management.azure.com/schemas/2016-04-30-preview/Microsoft.Compute.json#/resourceDefinitions/extensions"},
+                  { "$ref": "http://schema.management.azure.com/schemas/2016-06-01/Microsoft.Network.json#/resourceDefinitions/publicIPAddresses" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2016-06-01/Microsoft.Network.json#/resourceDefinitions/virtualNetworks" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2016-06-01/Microsoft.Network.json#/resourceDefinitions/loadBalancers" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2016-06-01/Microsoft.Network.json#/resourceDefinitions/networkSecurityGroups" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2016-06-01/Microsoft.Network.json#/resourceDefinitions/networkInterfaces" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2016-06-01/Microsoft.Network.json#/resourceDefinitions/routeTables" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2016-07-01/Microsoft.Network.json#/resourceDefinitions/publicIPAddresses" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: Seems this one is a invalid path. Should be 2016-07-01-preview?

**Comment by EvgenyAgafonchikov** [comment]
These PRs update schemas for different networking resources. I've merged

**Comment by DeepakRajendranMsft** [comment]
@TianoMS  @EvgenyAgafonchikov i have signed off on this

File: schemas/2016-07-01-preview/Microsoft.Network.json [hunk 2]
```diff
@@ -0,0 +1,1111 @@
...
+      "type": "object",
+      "properties": {
+        "type": {
+          "enum": [
+            "Microsoft.Network/loadBalancers"
+          ]
+        },
+        "apiVersion": {
+          "enum": [
+            "2016-07-01-preview"
```
Comment (DeepakRajendranMsft) [inline schemas/2016-07-01-preview/Microsoft.Network.json]: @TianoMS please wait till this change is done by Evgeny

Comment (TianoMS) [inline schemas/2016-07-01-preview/Microsoft.Network.json]: will hold on with merging.

**Comment by EvgenyAgafonchikov** [comment]
Updated (all versions w/o -preview). Fixed merge conflicts

**Comment by TianoMS** [comment]
@EvgenyAgafonchikov I'll do a bit more review before merging.

**Pull request merged**